lib/developer_portal/app/views/shared/cms/_toolbar.html.slim

```diff
@@ -1,120 +1,72 @@
-
-- if site_account.cms_toolbar_intro_visible?
-  javascript:
-    (function(){
-      var added_style = $();
-
-      var add_style_to_iframe = function () {
-        var style = $(this).find('option:selected').data('snippet');
-        var frame = frames['developer-portal'];
-        var doc;
-
-        if (frame.document == undefined)
-          doc = (frame.contentWindow || frame.contentDocument);
-        else
-          doc = frame;
-
-        added_style.remove();
-
-        if(!style) return;
-
-        added_style = $('<style>' + style + '</style>');
-
-        $('body', doc.document || doc).append(added_style);
-      };
-
-      $('iframe#developer-portal').on('load', add_style_to_iframe);
-      $('#theme-picker')
-              .on('change', add_style_to_iframe)
-              .on('change', function () {
-                var picker = $(this);
-
-                var snippet = picker.find('option:selected').data('snippet');
-                var style = "<style>" + snippet + '\n' + "</style>";
-                var explain = '<!-- Copy & paste this snippet into your template called main layout to make this change permanent -->';
-                var text = snippet ? explain + '\n\n' + style : '';
-
-                $('#theme-snippet')
-                        .toggle(text.length > 0)
-                        .find('textarea').val(text);
-
-              });
-    }());
+= javascript_packs_with_chunks_tag 'dev_portal_cms_toolbar'
+= stylesheet_link_tag "font-awesome" unless Rails.env.test?
+
+div class="pf-c-drawer pf-m-inline #{'pf-m-expanded' unless hidden}"
+  div class='pf-c-drawer__main'
+
+    div class="pf-c-drawer__content"
+      div class="pf-c-drawer__body"
+        iframe id="developer-portal" srcdoc=original_page_source
+
+    - if @page.respond_to?(:draft) && draft
+      button class="pf-c-button pf-m-plain" type="button" title="Toggle toolbar" id='cms-toolbar-toggle'
+        i class="fa fa-caret-square-o-right" aria-hidden="true"
+
+    div class="pf-c-drawer__panel" id="cms-toolbar"
+      div class="pf-c-drawer__panel-main"
+        div class="pf-c-drawer__body"
+          div class="pf-c-drawer__head"
+            nav class="pf-c-nav pf-m-tertiary"
+              ul class="pf-c-nav__list"
+                - if @page.respond_to?(:draft)
+                  li class="pf-c-nav__item"
+                    = link_to 'Draft', url_for(request.query_parameters.merge(cms: :draft)),
+                                       class: "pf-c-nav__link #{'pf-m-current' if draft}"
+                  li class="pf-c-nav__item"
+                    = link_to 'Published', url_for(request.query_parameters.merge(cms: :published)),
+                                           class: "pf-c-nav__link #{'pf-m-current' unless draft}"
+                - else
+                  li class="pf-c-nav__item"
+                    a href="#" class="pf-c-nav__link pf-m-current" aria-current="page"
+                      'Static Built-in Page
```

Comment:
@josemigallas How to reach this branch? :thinking:

Reply:
I think this is some kind of fail-safe they added long time ago (~9 year). I don't think it's possible to reach that branch nowadays, unless we override the condition manually.

Here it is when they added the first condition (April 2013): https://github.com/3scale/system/commit/f24dde7d55e948a9b9fa7b6905469562ad305f38#diff-168a5558ac27f8ed51aadf403fea552095bcfe42ef3c1d73d91e3cf84c3d3455

And here the else clause 5 months after: https://github.com/3scale/system/commit/bc737552febc8dfba05904f952b7d1add2f72e7a